src/components/CytoscapeGraph/graphs/GraphStyles.ts

```diff
@@ -367,7 +367,7 @@ export class GraphStyles {
       }
 
       const mtlsPercentage = edgeData.isMTLS;
-      if (cyGlobal.showSecurity && mtlsPercentage && mtlsPercentage >= 0) {
+      if (cyGlobal.showSecurity) {
         if (mtlsPercentage > 0 && !cyGlobal.mtlsEnabled) {
           content = `${EdgeIconMTLS} ${content}`;
         } else if (mtlsPercentage < 100 && cyGlobal.mtlsEnabled) {
```

Comment:
I agree with the change of the outer conditional, we should drop down when the Security display option is enabled.   I think after that for clarity we should have a conditional on !cyGlobal.mtlsEnabled because that determines how to handle the actual mtlsPercentage and should clear up any issue showing the open lock badge.  Maybe something like:

```
 if (cyGlobal.showSecurity) {
    if (cyGlobal.mtlsEnabled) {
         if (!mtlsPercentage || mtlsPercentage < 100) {
             content = `${EdgeIconDisabledMTLS} ${content}`;
         }
    } else {
        if (mtlsPercentage && mtlsPercentage > 0) {
          content = `${EdgeIconMTLS} ${content}`;
        } 
    }
}
```

Does that look good?

Reply:
yep, code looks better this way.
There is one more problem we have to approach here: what when there is no traffic?

![Screenshot of Kiali Console (67)](https://user-images.githubusercontent.com/613814/81953200-b000c780-9607-11ea-99e5-c40a437d5d1f.png)

So my proposal is to not show any badge if there isn't traffic. The result is:

![Screenshot of Kiali Console (68)](https://user-images.githubusercontent.com/613814/81953649-3a492b80-9608-11ea-950e-87256affdc02.png)

Reply:
@xeviknal +1, the lock badging is supposed to indicate information about the traffic, so no traffic should mean no badging.